scripts/SREP/update-upgrade-policy-state/script.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# The managed script update_upgrade_policy provides the ability to update the state
+# of the current OpenShift cluster's upgrade policy. It fetches the current state and
+# compares it to the desired state provided, updating it if necessary.
+
+# Usage: 
+# ./script.sh DESIRED_STATE=scheduled 
+# `scheduled` is the desired state you want to set for the cluster's upgrade policy.
+# 
+# Possible values of the upgrade policy state can include "started", "scheduled", "cancelled" etc. 
+# Ensure that you provide a state recognized by your OpenShift Cluster Manager instance.
+
+# Defined Functions
+
+# Fetch the unique identifier (UUID) of the current OpenShift cluster.
+get_cluster_uuid() {
+    oc get clusterversion version -o json | jq -r '.spec.clusterID'
+}
+
+# Using the cluster's UUID, fetch the cluster's ID.
+get_cluster_id() {
+    ocm list clusters --parameter search="external_id is '${1}'" --columns id --padding 20 --no-headers
+}
+
+# Get upgrade policies associated with the specified cluster ID.
+get_policy_response() {
+    ocm get "/api/clusters_mgmt/v1/clusters/${1}/upgrade_policies/"
+}
+
+# Count the number of upgrade policies available for the current cluster.
+get_policy_count() {
+    echo "$1" | jq '.size'
+}
+
+# Fetch the UUID of the first available upgrade policy for the cluster.
+get_policy_uuid() {
+    echo "$1" | jq -r '.items[0].id'
+}
+
+# Fetch the current state of the specified upgrade policy.
+get_current_state() {
+    ocm get "$1" | jq -r '.value'
+}
+
+# Update the state of the specified upgrade policy to the desired state.
+update_upgrade_policy() {
+    local ocm_slug="$1"
+    local desired_state="$2"
+    local payload="{ \"value\": \"${desired_state}\", \"description\": \"Manually updated by SRE\" }"
+    ocm patch "${ocm_slug}" --body "${payload}" --type application/json
+}
+
+# Main logic
+
+# Parse provided arguments
+for arg in "$@"; do
+    case "$arg" in
+        DESIRED_STATE=*)
+            DESIRED_STATE="${arg#DESIRED_STATE=}"
+            ;;
+        *)
+            echo "Invalid argument: $arg"
+            exit 1
+            ;;
+    esac
+done
+
+# Validate that a desired state argument is provided
```

Comment:
```suggestion
# Validate that a desired state argument is provided
```
Though , I think it'll be mostly handled already but can we also have another clean clear check on checking whether the DESIRED STATE passed by the user is a valid one?

Reply:
Yeah, we can purge all the `CSSRE` references, only `LPSRE` stands true

Reply:
Thanks for your time to review. It's a good catch. 

Since it's an initial implementation based on my understanding in the SOP,  It's not quite clear mentioning the expected upgrade_policy state we want to forcefully set and any dependencies or restrictions for the current state to be updated to the desired state.

For now, I will define the DESIRED_STATE as one of `started`, `scheduled` or `cancelled` and update the code accordingly.